Wrappers/Python/cil/optimisation/operators/BlockOperator.py

```diff
@@ -135,26 +136,43 @@ def get_item(self, row, col):
         index = row*self.shape[1]+col
         return self.operators[index]
     
-    def norm(self, **kwargs):
-        '''Returns the norm of the BlockOperator
-
-        if the operator in the block do not have method norm defined, i.e. they are SIRF
-        AcquisitionModel's we use PowerMethod if applicable, otherwise we raise an Error
+    def norm(self):
+        '''Returns the square root of the sum of the norms of the individual operators in the BlockOperators 
+        '''
+        return numpy.sqrt(numpy.sum(numpy.array(self.norms())**2))    
+    
+    def norms(self, ):
+        '''Returns a list of the individual norms of the Operators in the BlockOperator
         '''
-        norm = []
+        norms= []
         for op in self.operators:
-            if hasattr(op, 'norm'):
-                norm.append(op.norm(**kwargs) ** 2.)
-            else:
-                # use Power method
-                if op.is_linear():
-                    norm.append(
-                            LinearOperator.PowerMethod(op, 20)[0]
-                            )
-                else:
-                    raise TypeError('Operator {} does not have a norm method and is not linear'.format(op))
-        return numpy.sqrt(sum(norm))    
+            try:
+                norms.append(op.norm())
+            except:
+                    raise TypeError('Operator {} does not have a norm method'.format(op))
+        return norms
```

Comment:
```python
norms = [op.norm() for op in self.operators]
```
If norm doesn't exit python will raise an error something along the lines of:
```
AttributeError: type object 'Operator' has no attribute 'norm'
```
Which I think should be sufficient, we could catch it and add more information if needed.

Reply:
Sounds good - have changed this

Reply:
And now changed the list comprehension too